linera-chain/src/data_types.rs

```diff
@@ -726,12 +734,12 @@ impl CertificateValue {
     /// Creates a `HashedCertificateValue` by hashing `self`. No hash checks are made!
     pub fn with_hash(self) -> HashedCertificateValue {
         let hash = CryptoHash::new(&self);
-        HashedCertificateValue { value: self, hash }
+        HashedCertificateValue::unchecked_new(self, hash)
```

Comment:
Maybe `Hashed` should have a `new` or `compute` function that computes the hash?